### PR TITLE
Diagnostic Tool to show node movements between ds epochs

### PR DESCRIPTION
**Pull request by robin-thomas**
## Description

This tool reads the diagnostic DB and outputs a CSV formatted file, which shows the movement of an ip node (whether it's in ds committee or if it's in a shard, the shard index and the index within the shard) between various ds epochs

Make sure that the tool is run when the lookup node is not running.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/diagnostic/getnetworkhistory.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    output += row + "\n";
+  }
+}
+
+int main(int argc, char** argv) {
+  if (argc < 2) {
+    std::cout << "[USAGE] " << argv[0]
+              << " <output csv filename> <epoch width required>" << std::endl;
```
Comment (ansnunez) [inline src/diagnostic/getnetworkhistory.cpp]: I think `<epoch width required>` is unused and can be removed?

Comment (ansnunez) [reply src/diagnostic/getnetworkhistory.cpp]: I think it would be good to include the location of the database as a parameter

**Comment by robin-thomas** [comment]
@Gnnng 
> It seems the Travis build had some issues. I will check further on this PR once it's been fixed.

Yeah. It was Travis issue.

> One comment for this PR is it's better to move this utility to `src/cmd`.

I did have that discussion with Antonio. Since there are plans to make many diagnostic tools, thought it would be better to place it not in `cmd`.

File: src/depends/libDatabase/LevelDB.cpp [hunk 2]
```diff
@@ -87,21 +82,13 @@ LevelDB::LevelDB(const string & dbName, const string & subdirectory)
     leveldb::DB* db;
     leveldb::Status status;
 
-    if(m_subdirectory.empty())
+    string db_path = diagnostic ? ((m_subdirectory + (m_subdirectory.back() == '/' ? "" : "/")) + PERSISTENCE_PATH) : ("./"  + PERSISTENCE_PATH + (m_subdirectory.empty() ? "" : "/" + m_subdirectory));
```
Comment (ansnunez) [inline src/depends/libDatabase/LevelDB.cpp]: To simplify this long statement, maybe remove the need for checking `m_subdirectory.back` by making sure getnetworkhistory removes any trailing slash before calling this constructor.

Also, might be good to put a comment here about why we handle diagnostic differently.  You can say that the bool is only ever true when used by the tool, and that the non-diagnostic path is preserved from the original code.

File: src/diagnostic/getnetworkhistory.cpp (hunk 1 above)
Comment (robin-thomas) [reply src/diagnostic/getnetworkhistory.cpp]: Done!

File: src/depends/libDatabase/LevelDB.cpp (hunk 2 above)
Comment (robin-thomas) [reply src/depends/libDatabase/LevelDB.cpp]: Done!

**Pull request merged**